homeassistant/helpers/entity.py

```diff
@@ -273,6 +276,16 @@ def async_write_ha_state(self):
     @callback
     def _async_write_ha_state(self):
         """Write the state to the state machine."""
+        if self.registry_entry and self.registry_entry.disabled_by:
+            if not self._disabled_reported:
+                self._disabled_reported = True
+                _LOGGER.warning(
+                    "Entity %s is incorrectly being triggered for updates while it is disabled. This is a bug in the %s integration.",
+                    self.registry_entry.entity_id,
```

Comment:
Why not use `self.entity_id`?

Reply:
I thought that it would still be set, even if the entry was disabled. I was wrong

https://github.com/home-assistant/home-assistant/blob/bff5b00a09dcbe15c07ddcbce705c9487c0afcfb/homeassistant/helpers/entity_platform.py#L352-L362

I will update entity platform to always set entity ID and config entry, since we guard the writes now.